### ISSUE TITLE
advanced plug-in model editor - improvements (Trac #754)

### ISSUE THREAD

**Issue by RichardHeenan** [SasView/sasview#870]
(1) Minor chicken & egg situation here.
The "advanced" plug-in model editor only opens successfully compiled models. Even if you close an opened file, the file/open is greyed out so you cannot open any other file unless you instead go to the "tools" drop down and use the full python shell.

(2) Note also that the latter when opening mymodel.py also opens associated mymodel.c, but the advanced model editor does not.

(3) Possibly change custom model editor items to rename "new" to "new simple" and add "new advanced" to allow to open any file in plug-in models directory (avoiding temptation for users to edit a built in model under sasview!) [The simple editor is very good!]

(4) "delete" could put up an "are you sure" box as it really does delete the files, not just remove the model from sasview.

(5) can "check model" also perform unit tests rather than having to load and run separate python code?

Migrated from http://trac.sasview.org/ticket/754
```json
{
    "status": "closed",
    "changetime": "2018-11-27T18:16:26",
    "_ts": "2018-11-27 18:16:26.600169+00:00",
    "description": "(1) Minor chicken & egg situation here.\nThe \"advanced\" plug-in model editor only opens successfully compiled models. Even if you close an opened file, the file/open is greyed out so you cannot open any other file unless you instead go to the \"tools\" drop down and use the full python shell.\n\n(2) Note also that the latter when opening mymodel.py also opens associated mymodel.c, but the advanced model editor does not.\n\n(3) Possibly change custom model editor items to rename \"new\" to \"new simple\" and add \"new advanced\" to allow to open any file in plug-in models directory (avoiding temptation for users to edit a built in model under sasview!) [The simple editor is very good!]\n\n(4) \"delete\" could put up an \"are you sure\" box as it really does delete the files, not just remove the model from sasview.\n\n(5) can \"check model\" also perform unit tests rather than having to load and run separate python code?",
    "reporter": "richardh",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2016-10-10T14:21:14",
    "component": "SasView",
    "summary": "advanced plug-in model editor - improvements",
    "priority": "major",
    "keywords": "",
    "milestone": "SasView 5.0.0",
    "owner": "",
    "type": "defect"
}
```

**Comment by pkienzle** [comment]
Trac update at `2016/10/10 14:35:14`: **pkienzle** commented:

Re: (5), `sas.sasgui.perspectives.calculator.pyconsole.check_model` could instead call `sasmodels.model_test.run_one`.  The `run_one` function will need to be updated so that output is sent to a string and returned from the function using

```text
    ...
    stream = _WritelnDecorator(StringIO()))
    ...
    result = stream.get_value()
    stream.close()
    return result
```

**Comment by smk78** [comment]
Trac update at `2016/10/10 15:13:11`: **smk78** commented:

In changeset 8a631370b7b82ba2689d158abfc3faf1840f39b5:

```text
#!CommitTicketReference repository="sasview" revision="8a631370b7b82ba2689d158abfc3faf1840f39b5"
Updated documentation and images for the Python shell/editor. Refs #870
```

**Comment by butlerpd** [comment]
Trac update at `2016/10/11 18:23:48`: 
* **butler** commented:

really need to fix this but given resource constraints and priority of other tickets reluctantly moving to 4.2 along with related ticket #791


* **butler** changed milestone from "SasView 4.1.0" to "SasView 4.2.0"

**Comment by butlerpd** [comment]
Trac update at `2017/10/26 12:45:19`: **butler** changed milestone from "SasView 4.2.0" to "SasView 5.0.0"

**Comment by rozyczko** [comment]
Trac update at `2018/11/22 08:18:14`: **piotr** commented:

None of the issues in the bug description applies to the new plugin editor/manager. This can be closed, I think.

**Comment by butlerpd** [comment]
Trac update at `2018/11/23 15:26:31`: **butler** commented:

Hummm... indeed it looks like most of these were in fact fixed by 4.2.  The only question I guess is whether the advanced editor should read **any** file in the directory or just valid model files? Am tempted to say we are done here but actually would need to test the latest 5.0 version I guess.  Should this be discussed at next conf call?

**Comment by butlerpd** [comment]
Trac update at `2018/11/27 18:16:26`: 
* **butler** commented:

This last is also fixed in 5.0 according to Piotr so we can close this ticket.


* **butler** changed resolution from "" to "fixed"
* **butler** changed status from "new" to "closed"
